### PR TITLE
LWDEV-10485 Add deduplication to Trade Volume Monitoring Service

### PR DESCRIPTION
**Pull request by barsukovskii**

### CODE REVIEW

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt [hunk 1]
```diff
@@ -1,16 +1,64 @@
...
+    companion object {
+        private val LOGGER = LoggerFactory.getLogger(EventDeduplicationServiceImpl::class.java.name)
+    }
+
+    private val lock = ReentrantLock()
+    private var initialized = false
+    private var sequenceNumbers: MutableSet<Long> = HashSet()
+    private var prevSequenceNumbers: Set<Long>? = null
+
+    @Synchronized
```
Comment (olpapchenko) [inline src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt:26]: suggest to use only one lock here `lock` as lock inside the lock is deadlock prone

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt [hunk 2]
```diff
@@ -1,16 +1,64 @@
...
+    fun init() {
+        if (initialized) {
+            throw ApplicationException("Already initialized")
+        }
+        initialized = true
```
Comment (olpapchenko) [inline src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt]: set this var at the end

**Review by olpapchenko — APPROVED** [review]

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt (hunk 1 above)
Comment (barsukovskii) [reply src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt:26]: Decided to leave it as it is as these locks have different purposes

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt (hunk 2 above)
Comment (barsukovskii) [reply src/main/kotlin/com/lykke/trade/volume/monitoring/service/process/impl/EventDeduplicationServiceImpl.kt]: fixed

**Pull request merged**